### ISSUE TITLE
Redfish Query for Host OS Network Interfaces

### ISSUE THREAD

**Issue by mtramont** [dell/iDRAC-Redfish-Scripting#281]
### Environment:  iDRAC 9 Firmware Version 7.00.00.00 with iSM in Running state.

Redfish REST Query "/redfish/v1/Systems/System.Embedded.1/EthernetInterfaces" from (GetOSNetworkInformationREDFISH.py) returns 0 members, however a CIM session query returns the correct data.

**Redfish REST Query "/redfish/v1/Systems/System.Embedded.1/EthernetInterfaces" Output:**

{
  "@odata.context": "/redfish/v1/$metadata#EthernetInterfaceCollection.EthernetInterfaceCollection",
  "@odata.id": "/redfish/v1/Systems/System.Embedded.1/EthernetInterfaces",
  "@odata.type": "#EthernetInterfaceCollection.EthernetInterfaceCollection",
  "Description": "Collection of Ethernet Interfaces for this System",
  "Members": [],
  "Members@odata.count": 0,
  "Name": "System Ethernet Interface Collection"
}

**Example CIM Session Output:**

DHCPEnabled       : true
DeviceDescription : vmk2
DeviceFQDD        :
IPv4Address       : 169.254.56.42
IPv4DHCPServer    :
IPv4DNSServer     :
IPv4Gateway       :
IPv4SubnetMask    : 255.255.0.0
IPv6AddrScope     : 2
IPv6Address       : fe80::250:56ff:fe6e:5479
IPv6DHCPServer    :
IPv6DNSServer     :
IPv6Gateway       :
IPv6PrefixLength  : 64
InstanceID        : iDRAC.Embedded.1#ServiceModule.1#OSLogicalNetwork.2
MACAddr           : 00-50-56-6E-54-79
Name              : vmk2
Status            : 0
Type              : 1
PSComputerName    : 10.10.1.1

**Comment by texroemer** [comment]
Hi @mtramont 

On the same server can you try this RACADM command and see if it reports correct OS network information (racadm gethostnetworkinterfaces)? If RACADM command works this is a known Redfish product bug which is currently being looked into by internal teams at Dell. 

Thanks
Tex

**Comment by mtramont** [comment]
Yes the RACADM "gethostnetworkinterfaces" function, as well as a CIM session query, reports the correct OS network information.  Which versions of Redfish are affected by this bug?

**Comment by texroemer** [comment]
Can you share what version of iDRAC and iSM is running on the server with the issue.

**Comment by mtramont** [comment]
iDRAC9, Lifecycle Controller 7.00.00.00, Dell EMC iDRAC Service Module Embedded Package v3.5.1 A00, iSM Installed Version on Host OS 5.2.0.0

**Comment by texroemer** [comment]
Thanks for the firmware version details, can you also confirm what server model you see the issue on, is it VxRail or a different model?

**Comment by mtramont** [comment]
Here are my test results against (1) PowerEdge Server and (2) VxRail Nodes.  The RedFish query against the PowerEdge server produces results, although the NIC device list is incomplete.  The RedFish query against 2 different VxRail node models produces zero results.  So does this issue only affect VxRail nodes?

Model:  PowerEdge R440
iDRAC Version:  9
iDRAC Firmware Version:  6.10.30.20 
iSM Installed Version on Host OS:  5.0.1.0
Redfish Query: Returns 1 NIC device:  vmk0
RACADM & CIM Query: Returns 3 NIC devices:  lo0, vmk0, vmk1

Model:  VxRail E560
iDRAC Version:  9
iDRAC Firmware Version:  7.00.00.00
iSM Installed Version on Host OS:  5.2.0.0
Redfish Query: Returns 0 NIC devices
RACADM & CIM Query: Returns 4 NIC devices: lo0, vmk0, vmk1, vmk2

Model:  VxRail P580N
iDRAC Version:  9
iDRAC Firmware Version:  5.10.30.201 
iSM Installed Version on Host OS:  4.1.0.0
Redfish Query: Returns 0 NIC devices
RACADM & CIM Query: Returns 5 NIC devices:  lo0, vmk0, vmk1/vusb0, vmk2, vmk3 (missing vmk4 & vmk5 which are also *not* displayed in iDRAC)

**Comment by texroemer** [comment]
Thanks for the config details this helps. So there's an issue with both PowerEdge and VxRail servers as you stated, PowerEdge will show partial details and VxRail will report no details. 

Still working with internal teams to get you an ETA on when the fix will be available (looks to be an iSM bug btw not iDRAC). For now you'll need to leverage RACADM as a workaround solution.

**Comment by texroemer** [comment]
@mtramont 

Wanted to give you an update, issue has been resolved in iDRAC and build will be available in March 2025, build number will be 7.20.10.50.

For the changes, when iSM is installed and running in ESXi, now URI "redfish/v1/Systems/System.Embedded.1/EthernetInterfaces" will report both hardware and OS network interfaces. 

Example output of the fix:

```
[root@localhost]# curl -k -X GET -u root:calvin https://192.168.0.120/redfish/v1/Systems/System.Embedded.1/EthernetInterfaces --insecure | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1038  100  1038    0     0   3359      0 --:--:-- --:--:-- --:--:--  3370
{
  "@odata.context": "/redfish/v1/$metadata#EthernetInterfaceCollection.EthernetInterfaceCollection",
  "@odata.id": "/redfish/v1/Systems/System.Embedded.1/EthernetInterfaces",
  "@odata.type": "#EthernetInterfaceCollection.EthernetInterfaceCollection",
  "Description": "Collection of Ethernet Interfaces for this System",
  "Members": [
    {
      "@odata.id": "/redfish/v1/Systems/System.Embedded.1/EthernetInterfaces/NIC.Embedded.1-1-1"
    },
    {
      "@odata.id": "/redfish/v1/Systems/System.Embedded.1/EthernetInterfaces/NIC.Embedded.2-1-1"
    },
    {
      "@odata.id": "/redfish/v1/Systems/System.Embedded.1/EthernetInterfaces/NIC.Integrated.1-2-1"
    },
    {
      "@odata.id": "/redfish/v1/Systems/System.Embedded.1/EthernetInterfaces/NIC.Integrated.1-1-1"
    },
    {
      "@odata.id": "/redfish/v1/Systems/System.Embedded.1/EthernetInterfaces/OSLogicalNetwork.1"
    },
    {
      "@odata.id": "/redfish/v1/Systems/System.Embedded.1/EthernetInterfaces/OSLogicalNetwork.2"
    },
    {
      "@odata.id": "/redfish/v1/Systems/System.Embedded.1/EthernetInterfaces/OSLogicalNetwork.3"
    }
  ],
  [Members@odata.count](mailto:Members@odata.count): 4,
  "Name": "System Ethernet Interface Collection"
}
```



Thanks
Tex
